hq/app/services/IamRemediationService.scala

```diff
@@ -96,12 +96,21 @@ class IamRemediationService(
   def disableUnrecognisedUsers()(implicit ec: ExecutionContext): Attempt[Unit] = {
     val result = for {
       config <- getIamUnrecognisedUserConfig(config)
+      // fetch and parse our stored Janus config to use the canonical source of "recognised" usernames
       s3Object <- getS3Object(securityS3Client, config.janusUserBucket, config.janusDataFileKey)
       janusData = JanusConfig.load(makeFile(s3Object.mkString))
       janusUsernames = getJanusUsernames(janusData)
+      // look up the credentials report from the cache service as our source of current IAM users
       accountCredsReports = getCredsReportDisplayForAccount(cacheService.getAllCredentials)
+      // determine the unrecognised users by comparing Janus usernames to the IAM users (and filter to allowed accounts)
       allowedAccountsUnrecognisedUsers = unrecognisedUsersForAllowedAccounts(accountCredsReports, janusUsernames, config.allowedAccounts)
-      _ <- Attempt.traverse(allowedAccountsUnrecognisedUsers)(disableAccountUsers(_, iamClients))
```

Comment:
instead of deeply nesting listing, disabling keys and removing passwords - we orchestrate it all from here